plugins/out_opensearch/opensearch.c

```diff
@@ -853,12 +904,149 @@ static int opensearch_error_check(struct flb_opensearch *ctx,
     return check;
 }
 
+static int put_index_template(struct flb_opensearch *ctx)
+{
+    int res;
+    size_t b;
+    struct flb_http_client *c = NULL;
+    flb_sds_t template;
+    struct flb_connection *u_conn;
+
+    u_conn = flb_upstream_conn_get(ctx->u);
+
+    if (!u_conn) {
+        return FLB_OS_RETRY;
+    }
+
+    template = flb_file_read(ctx->template_file);
+
+    /*
+     * template does not exist so retry
+     */
+    if(flb_sds_len(template) == 0) {
+        flb_errno();
+        flb_plg_debug(ctx->ins, "failed to read template file");
+        return FLB_OS_RETRY;
+    }
+
+    c = flb_http_client(u_conn, FLB_HTTP_PUT, ctx->template_uri,
+                        template, flb_sds_len(template), NULL,
+                        0, NULL, 0);
+
+    if (ctx->http_user && ctx->http_passwd) {
+        flb_http_basic_auth(c, ctx->http_user, ctx->http_passwd);
+    }
+    flb_http_add_header(c, "Content-Type", 12, "application/json", 16);
+
+    res = flb_http_do(c, &b);
+    if(res != 0) {
+        flb_http_client_destroy(c);
+        flb_upstream_conn_release(u_conn);
+        flb_sds_destroy(template);
+        return FLB_OS_RETRY;
+    }
+    if(c->resp.status != 200 && c->resp.status != 201) {
+        flb_plg_debug(ctx->ins, "HTTP Status=%i URI=%s", c->resp.status, ctx->template_uri);
+        flb_http_client_destroy(c);
+        flb_upstream_conn_release(u_conn);
+        flb_sds_destroy(template);
+        return FLB_OS_RETRY;
+    }
+    flb_http_client_destroy(c);
+    flb_upstream_conn_release(u_conn);
+    flb_sds_destroy(template);
+
+    return FLB_OS_SUCCESS;
+}
+
+static int template_exists(struct flb_opensearch *ctx) {
+    struct flb_connection *u_conn;
+    int res;
+    size_t b;
+    struct flb_http_client *c = NULL;
+    u_conn = flb_upstream_conn_get(ctx->u);
+
+    if (!u_conn) {
+        return FLB_OS_RETRY;
+    }
+
+    c = flb_http_client(u_conn, FLB_HTTP_GET, ctx->template_uri,
```

Comment:
`flb_http_client` can return `NULL`, please add the appropriate check here and in line 932.